organicmaps/organicmaps
# [android] use µG play-services-location in fdroid flavour

This is a FOSS re-implementation of google's play-services-location client library by the µG project. Because it's Apache-2.0 licensed, this ought to be permitted on F-droid but I'm not sure whether it'd be marked with the non-free net anti-feature.

Theoretically it should work just as well on devices with Google's GMS but I'd suggest testing this in the fdroid flavour first where there is no alternative and switch all flavors over if that works on actual user devices. Doing so would remove one major blocker towards achieving
https://github.com/organicmaps/organicmaps/issues/6906.

I have tested this on LineageOS with µG and on GrapheneOS with Google GMS. (You need to explicitly allow fused location from GMS and give it location permissions with the latter but it did work.)

For anyone else wanting to test this:

1. Ensure that you cannot get a GPS fix by checking GPSTest or GNSSLogger
2. Ensure that you are able to get a fused location using GNSSLogger
3. Install and open the f-droid build built from this branch
4. Wipe app data and open
5. Give it accurate location permission
6. If it asks you to download not just the world map but also your current area via a checkbox above the download button, it works.
